.circleci/config.yml

```diff
@@ -0,0 +1,214 @@
+          command: |
+            if [ "$(cat /tmp/go-version)" = "$(cat /usr/local/go/VERSION)" ] ; then
+              exit 0
+            fi
+            wget -O/tmp/go-master.tgz https://github.com/golang/go/archive/master.tar.gz
+            tar -C /tmp -zxf /tmp/go-master.tgz
+            mv /tmp/go-master /usr/local/go
+            export GOSRC=/usr/local/go
+            export GOROOT=$GOSRC
+            export GOROOT_BOOTSTRAP=/usr/local/go-stable
+            export GOBUILD=$GOSRC/src
+            cp /tmp/go-version $GOROOT/VERSION
+            cd $GOBUILD
+            ./make.bash
+      - save_cache:
+          key: |
+            {{ checksum "/tmp/go-version" }}
+          paths:
+            - /usr/local/go
+  test:
+    description: "Run the tests"
+    steps:
+      - run:
+          name: go test
+          command: |
+            mkdir -p /tmp/test-results/go
+            trap "go get -u github.com/jstemmer/go-junit-report && go-junit-report </tmp/go-test-output.txt > /tmp/test-results/go/test-report.xml" EXIT
+            go test -race -v -timeout 60s ./... | tee /tmp/go-test-output.txt
+      - store_test_results:
+          path: /tmp/test-results/go
+
+jobs:
+  success:
+    docker:
+      - image: alpine:latest
+    steps:
+      - run: echo yay
+
+  code_hygiene:
+    executor:
+      name: gofmt
+    working_directory: /go/src/github.com/stripe/veneur
+    steps:
+      - install
+      - checkout
+      - run:
+          name: go generate
+          command: |
+            export PATH=$PATH:/opt/protoc/bin
+            go generate
+      - run: "dep check"
+      - run:
+          name: "gofmt"
+          command: |
+            mv vendor /tmp/veneur-vendor
+            gofmt -w .
+            mv /tmp/veneur-vendor ./vendor
+      - run:
+          name: "check if any files changed"
+          command: |
+            git add .
+            env PAGER=cat git diff --cached
+            git diff-index --cached --exit-code HEAD
+
+  test_stable:
+    executor:
+      name: stable
+    working_directory: /go/src/github.com/stripe/veneur
+    steps:
+    - checkout
+    - test
+
+  test_previous:
+    executor:
+      name: previous
+    working_directory: /go/src/github.com/stripe/veneur
+    steps:
+      - checkout
+      - test
+
+  test_tip:
+    docker:
+      - image: golang:latest
+    working_directory: /go/src/github.com/stripe/veneur
+    environment:
+      GO111MODULE: auto
+    steps:
+      - fetch_and_build_latest_go
+      - checkout
+      - test
+
+workflows:
+  version: 2
+  continuous_integration:
+    jobs:
+      - code_hygiene
+      - test_stable
+      - test_previous
+      - test_tip
+      - success:
+          requires:
+          - code_hygiene
+          - test_stable
+          - test_previous
+          - test_tip
+  scheduled_tests:
+    jobs:
+      - test_stable
+      - test_previous
+      - test_tip
+    triggers:
+      - schedule:
+          cron: 0 0 * * 0
```

Comment:
Is this UTC? 

If so, maybe we should pick a time that's during business hours.